File: typescript/src/n0169_majority_element.ts
export function majorityElement(numbers: number[]): number {
  let element = 0;
  let count = 0;
  for (const number of numbers) {
    if (count === 0) {
      element = number;
    }
    count += element === number ? 1 : -1;
  }
  return element;
}

File: rust/src/n0169_majority_element.rs
pub fn majority_element(numbers: &[i32]) -> i32 {
    let mut element = 0;
    let mut count = 0;
    for &number in numbers {
        if count == 0 {
            element = number;
        }
        count += if element == number { 1 } else { -1 };
    }
    element
}
